lib/src/layer/tile_layer.dart

```diff
@@ -317,6 +317,42 @@ class _TileLayerState extends State<TileLayer> with TickerProviderStateMixin {
     _update(null);
     _moveSub = widget.stream.listen((_) => _handleMove());
 
+    _initThrottleUpdate();
+  }
+
+  @override
+  void didUpdateWidget(TileLayer oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    var reloadTiles = false;
+
+    final oldUrl = oldWidget.options.urlTemplate ??
+        oldWidget.options?.wmsOptions?._encodedBaseUrl;
```

Comment:
Given that `options.wmsOptions` takes precedence once set - that should be the first to check, shouldn't it?

Reply:
If `wmsOptions` is present then `urlTemplate` should be `null` if is `not null` then programmer made a mistake. I will change code what you suggested however this shouldn't be fixed at this side.